solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -1609,12 +1611,41 @@ public void publish(final CoreDescriptor cd, final Replica.State state, boolean
       if (updateLastState) {
         cd.getCloudDescriptor().setLastPublished(state);
       }
-      overseerJobQueue.offer(Utils.toJSON(m));
+      DocCollection coll = zkStateReader.getCollection(collection);
+      if (forcePublish || sendToOverseer(coll, coreNodeName)) {
```

Comment:
why do we let forcePublish override us here? Won't this cause split state later?

Reply:
"split state" 
What do you mean by that?

Reply:
some aspects of state stored in the old place, some aspects stored in the new place. I'm still working on building a full mental model, so maybe this is a wrong question.